### PR TITLE
henson: simplify args logic with define_from_variant

### PR DESCRIPTION
**Pull request by anigmetov**
Closes #36285

Simplified, as suggested by @alalazo

### CODE REVIEW

**Comment by alalazo** [comment]
@anigmetov Thanks for the PR. FYI, next time you could push to your previous PR, and we can continue the discussion there :slightly_smiling_face:

**Review by alalazo — APPROVED** [review]
Waiting for @mrzv  review before merging

**Review by mrzv — APPROVED** [review]

**Pull request merged**
